### ISSUE TITLE
bpf_tcp_raw_gen_syncookie does not exist in Upstream Kernel

### ISSUE THREAD

**Issue by fionera** [Aniket999969/XDP-SYNPROXY#3]
In the code are function used that do not exist in the upstream Kernel. I looked for them and found them in a Patch Thread but they got modified before being picked. https://lore.kernel.org/bpf/20220124151340.376807-1-maximmi@nvidia.com/T/

These are the correct functions:
Helper | Kernel version | License | Commit
-- | -- | -- | --
BPF_FUNC_tcp_check_syncookie() | 5.2 |   | 399040847084
BPF_FUNC_tcp_gen_syncookie() | 5.3 |   | 70d66244317e
BPF_FUNC_tcp_raw_check_syncookie_ipv4() | 6.0 |   | 33bf9885040c
BPF_FUNC_tcp_raw_check_syncookie_ipv6() | 6.0 |   | 33bf9885040c
BPF_FUNC_tcp_raw_gen_syncookie_ipv4() | 6.0 |   | 33bf9885040c
BPF_FUNC_tcp_raw_gen_syncookie_ipv6() | 6.0 |   | 33bf9885040c

**Comment by scaetano84** [comment]
could you get the code to work? @fionera

**Comment by fionera** [comment]
No. The code ins this repo is mostly copied from examples/test from the linux bpf-next branch

**Comment by scaetano84** [comment]
@fionera Too bad, were you able to get TCP filtering through xdp? I would be interested in applying it on a server of mine.

**Comment by fionera** [comment]
I did build filtering and other things using XDP already, yes. Sadly none of it is open-source right now
